lib/HTTP/Cookies.pm

```diff
@@ -621,6 +621,50 @@ sub _normalize_path  # so that plain string compare can be used
     $_[0] =~ s/([\0-\x20\x7f-\xff])/sprintf("%%%02X",ord($1))/eg;
 }
 
+# deals with splitting values by ; and the fact that they could
+# be in quotes which can also have escaping.
+sub _split_text {
+    my $val = shift;
+    my @vals = grep { $_ ne '' } split(/([;\\"])/, $val);
```

Comment:
The issue is mainly that it can be hard to differentiate between `''` and `"`.  I'm not going to insist, though.  ;)